jwst/tweakreg/tweakreg_step.py

```diff
@@ -200,13 +202,91 @@ def process(self, input):
+                    use2dhist=self.use2dhist,
+                    tolerance=self.tolerance,
+                    xoffset=0.0,
+                    yoffset=0.0
+                )
+
+                # Set group_id to same value so all get fit as one observation
+                # The assigned value, 987654, has been hard-coded to make it
+                # easy to recognize when alignment to GAIA was being performed
+                # as opposed to the group_id values used for relative alignment
+                # earlier in this step.
+                for imcat in imcats:
+                    imcat.meta['group_id'] = 987654
+                    if 'REFERENCE' in imcat.meta['fit_info']['status']:
+                        del imcat.meta['fit_info']
+
+                # Perform fit
+                align_wcs(imcats,
+                          refcat=ref_cat,
+                          enforce_user_order=False,
+                          expand_refcat=False,
+                          minobj=self.minobj,
+                          match=tpmatch_gaia,
+                          fitgeom=self.fitgeometry,
+                          nclip=self.nclip,
+                          sigma=(self.sigma, 'rmse')
+                         )
+
+                # Also, update/create the WCS .name attribute with information
+                #  on this astrometric fit as the only record that it was successful
+                for imcat in imcats:
+                    if 'SUCCESS' in imcat.meta.get('fit_info')['status']:
+                        # NOTE: This .name attribute needs to be defined using a convention
+                        #       agreed upon by the JWST Cal Working Group.
+                        #       Current value is merely a place-holder based on HST conventions
+                        #       This value should also be translated to the FITS WCSNAME keyword
+                        #       IF that is what gets recorded in the archive for end-user searches
+                        imcat.meta['image_model'].meta.wcs.name = "FIT-LVL3-{}".format(self.gaia_catalog)
+
         for imcat in imcats:
             imcat.meta['image_model'].meta.cal_step.tweakreg = 'COMPLETE'
             # retrieve fit status and update wcs if fit is successful:
-            fit_info = imcat.meta.get('fit_info')
-            if fit_info['status'] in 'SUCCESS':
+            if 'SUCCESS' in imcat.meta.get('fit_info')['status']:
                 imcat.meta['image_model'].meta.wcs = imcat.wcs
 
+                """
+                # Also update FITS representation in input exposures for
+                # subsequent reprocessing by the end-user.
+                # Not currently enabled, but may be requested later...
+                gwcs_header = imcat.wcs.to_fits_sip(max_pix_error=0.1,
+                                                max_inv_pix_error=0.1,
+                                                degree=3,
+                                                npoints=128)
```

Comment:
How could we update headers when a number of questions of how to do it are not answered? The first that comes to mind is that SIP writes out CD matrix. JWST uses and has in the headers a PC matrix. Would this write both and what does this mean?

Reply:
Yeah, this whole block has been commented out with `"""`.  (Easy to miss).  We discussed this above in a previous round of review. Agree we don't want to do this right now.